ee/src/main/java/org/jboss/as/ee/subsystem/GlobalDirectoryResourceDefinition.java

```diff
@@ -0,0 +1,175 @@
+import static org.jboss.as.ee.subsystem.EeCapabilities.EE_GLOBAL_DIRECTORY_CAPABILITY;
+import static org.jboss.as.ee.subsystem.EeCapabilities.PATH_MANAGER_CAPABILITY;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AbstractRemoveStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.CapabilityServiceBuilder;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.PersistentResourceDefinition;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.controller.services.path.PathManager;
+import org.jboss.as.ee.logging.EeLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.msc.Service;
+import org.jboss.msc.service.ServiceController;
+
+/**
+ * The resource definition for global-directory child resource in EE subsystem.
+ *
+ * @author Yeray Borges
+ */
+public class GlobalDirectoryResourceDefinition extends PersistentResourceDefinition {
+
+    static SimpleAttributeDefinition PATH = create(ModelDescriptionConstants.PATH, ModelType.STRING, false)
+            .setAllowExpression(true)
+            .addArbitraryDescriptor(FILESYSTEM_PATH, new ModelNode(true))
+            .setRestartJVM()
+            .build();
+
+    static SimpleAttributeDefinition RELATIVE_TO = create(ModelDescriptionConstants.RELATIVE_TO, ModelType.STRING, true)
+            .setAllowExpression(false)
+            .setRestartJVM()
+            .build();
+
+    static final SimpleAttributeDefinition[] ATTRIBUTES = new SimpleAttributeDefinition[]{PATH, RELATIVE_TO};
+
+    private static final AbstractAddStepHandler ADD = new GlobalDirectoryAddHandler();
+    private static final AbstractRemoveStepHandler REMOVE = new GlobalDirectoryRemoveHandler();
+
+    public static final GlobalDirectoryResourceDefinition INSTANCE = new GlobalDirectoryResourceDefinition();
+
+    public GlobalDirectoryResourceDefinition() {
+        super(new SimpleResourceDefinition.Parameters(PathElement.pathElement(EESubsystemModel.GLOBAL_DIRECTORY), EeExtension.getResourceDescriptionResolver(EESubsystemModel.GLOBAL_DIRECTORY))
+                .setAddRestartLevel(OperationEntry.Flag.RESTART_JVM)
```

Comment:
This was left as a mistake. At first glance, I thought the only reliable way to be able to unload the module created by the configuration of a global-directory was to execute a full server restart. 

After your comment, I've been checking how it behaves using a reload and since all the services are removed, the module created is also unloaded. @tterem I've changed the tests to use a reload instead of a full server restart, let's wait for CI.